### DISCUSSION TITLE
Migrating from other version control services

### DISCUSSION THREAD

**Discussion by areynold** [uillinois-community/uillinois-community.github.io#28] [Q&A]
What tools are folks using to import repositories and issues from other version control services?

We had several repos hosted on Engineering IT's GitLab server. I had assumed I would be able to use GitHub's import tool, but my experience mirrored [this writeup](https://jdonkervliet.com/2020/08/24/gitlab-to-github.html) pretty closely--simple public repos worked fine, while those with active participation needed a lot of manual work.

I ended up using [piceaTech/node-gitlab-2-github](https://github.com/piceaTech/node-gitlab-2-github) (mentioned in the above article), which worked pretty well. It failed to migrate some attachments (e.g., screenshots in issues), but it did log those errors. We didn't have many attachments to begin with, and most were on closed issues, so it wasn't a deal breaker for us.

**Comment by edthedev** [comment]
I have previously used [roughly this recipe](https://stackoverflow.com/questions/79165/how-do-i-migrate-an-svn-repository-with-history-to-a-new-git-repository) to migrate history from SVN repositories.
